CTCTRADERS-2848: testOnly route to drop mongo collection.

jackheslop96 opened the pull request.

DavidGregory084 commented on conf/application.conf:
```diff
@@ -51,6 +51,7 @@ play.modules.enabled += "uk.gov.hmrc.mongo.play.PlayMongoModule"
 play.modules.enabled += "runtime.IOModule"
 play.modules.enabled += "runtime.SystemClockModule"
 play.modules.enabled += "workers.BalanceRequestUpdateWorkerModule"
+play.modules.enabled += "play.modules.reactivemongo.ReactiveMongoModule"
```
Please do not do this - this service uses hmrc-mongo because reactivemongo is deprecated

DavidGregory084 commented on project/AppDependencies.scala:
```diff
@@ -11,6 +10,7 @@ object AppDependencies {
   val hmrcMongoVersion  = "0.55.0"
 
   val compile = Seq(
+    "org.reactivemongo"      %% "play2-reactivemongo"       % "0.20.13-play28",
```
Please do not do this - this service uses hmrc-mongo because reactivemongo is deprecated

DavidGregory084 commented on app/controllers/testOnly/TestOnlyMongoController.scala:
```diff
@@ -0,0 +1,47 @@
+          case false => InternalServerError
+        }
+      )
+  }
```
I think the best way to do this in hmrc-mongo is to inject the `PlayMongoComponent` and use the `database` that it provides - see e.g. http://mongodb.github.io/mongo-java-driver/4.2/driver-scala/tutorials/databases-collections/#drop-a-collection

DavidGregory084 replied:
Oh and https://github.com/hmrc/hmrc-mongo/blob/main/hmrc-mongo-common/src/main/scala/uk/gov/hmrc/mongo/MongoComponent.scala

DavidGregory084 commented on conf/testOnlyDoNotUseInAppConf.routes:
```diff
@@ -3,11 +3,14 @@
 # Failing to follow this rule may result in test routes deployed in production.
 
++ nocsrf
+DELETE     /transit-movements-guarantee-balance/test-only/drop-mongo-collection       controllers.testOnly.TestOnlyMongoController.dropMongoCollection()
```
Very small thing, but would it be possible to change this route to `/transit-movements-guarantee-balance/test-only/balances`?
I think the intent to drop the collection is already clear from the `DELETE` method, and this endpoint operates on balance requests.

DavidGregory084 commented on app/controllers/testOnly/TestOnlyMongoController.scala:
```diff
@@ -0,0 +1,47 @@
+import scala.concurrent.ExecutionContext
+
+class TestOnlyMongoController @Inject() (
+  override val messagesApi: MessagesApi,
```
Is this `override` needed?

jackheslop96 commented:
@DavidGregory084 thank you for reviewing this yesterday evening it's very much appreciated and the comments were extremely insightful. I had been lazy and just copied some code from transit-movements-trader-at-destination. I hope my latest commit has addressed all of your comments - if there are any further issues let me know :)

DavidGregory084 reviewed (approved):
Yep looks good to me! 💯
